db/column_family.cc

```diff
@@ -507,7 +519,19 @@ ColumnFamilyData::ColumnFamilyData(
       queued_for_compaction_(false),
       prev_compaction_needed_bytes_(0),
       allow_2pc_(db_options.allow_2pc),
-      last_memtable_id_(0) {
+      last_memtable_id_(0),
+      db_paths_registered_(false) {
+  if (id_ != kDummyColumnFamilyDataId) {
+    Status s = ioptions_.env->RegisterDbPaths(GetDbPaths());
```

Comment:
If possible, avoid potentially expensive operation like this in side constructors. But I assume that you already weighed in those considerations and feel this is the cleanest way we can get.

Do we have access to FileSystem object here? If we have, it will be a good idea to call FileSystem rather than Env here.

Reply:
This is the cleanest way, the downside is that this might be called with db mutex held. In `Open`, it's not big deal since no other threads will start writing during Open, but for dynamically creating column families, this will bring performance down when there are a lot of data paths for a CF.

Reply:
replaced env with fs.

Reply:
I cannot find a cleaner way right now, but I'll try to remove the RegisterDbPaths call from the code path with db mutex held in another PR.

Reply:
left a TODO in the code